source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,236 @@
+database results while holding only a subset of them in memory at a
+given time. Read operations that match multiple documents use a cursor
+to return those documents in batches as opposed to all at once.
+
+Sample Cursor
+~~~~~~~~~~~~~
+
+The following sections use a ``cursor`` variable, which is a cursor that
+retrieved all the documents in a collection with the ``Find()`` function.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+.. important::
+
+   A cursor is not goroutine safe. Do not use the same cursor in
+   multiple goroutines at the same time.
+
+Retrieve All Documents
+----------------------
+
+To populate an array with all of your query results, use the ``All()``
+method:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor all
+   :end-before: end cursor all
+
+.. important:: Memory
+
+   This method crashes your application if your query retrieves a number
+   of documents larger than the size of your array. If this occurs,
+   :ref:`retrieve the documents individually <individual-documents-golang>`.
+
+.. _individual-documents-golang:
+
+Retrieve Documents Individually
+-------------------------------
+
+To retrieve documents from your cursor individually, use the
+``Next()`` method.
+
+This method blocks the current goroutine. That means the function
+returns a document if each of the following is true:
+
+- A document is currently or will later be available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. tip::
+
+   Iterating through your documents individually removes the possibility
+   of crashing your application and allows you to :ref:`access a cursors
+   elements <cursor-elements-golang>`.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor next
+   :end-before: end cursor next
+
+.. note::
+   
+   If you previously used the cursor to iterate some documents, the
+   driver excludes those documents when you iterate again. 
+
+Documents Only in the Cursor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To attempt retrieving a document from your cursor, use the ``TryNext()``
+method.
+
+This method is unable to block the current goroutine. That means the
+function returns a document if each of the following is true:
+
+- A document is currently available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor try next
+   :end-before: end cursor try next
+
+.. note::
+   
+   If you previously used the cursor to iterate some documents, the
+   driver excludes those documents when you iterate again. 
+
```

Comment:
Suggestion: After looking at the API docs for the [`TryNext()` method](https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.7.2/mongo#Cursor.TryNext) it seems like this is only meant to be used for watching for changes in capped collections. Could the section be tweaked to reflect this?

Question: What is the exact mechanism by which a tailable/non-tailable cursor is created? I can't find this information. Are all cursors created by the go driver tailable or only some?

Rough draft :

```suggestion
Wait For Documents After The Cursor Is Exhausted
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If you need to wait for more documents after your cursor is exhausted, use a
**tail-able cursor**. A tail-able cursor is a cursor that remains open after it is
exhausted and watches for additional documents that match your query.

.. important:: Only Available for Capped Collections

   Tailable cursors are only available for :manual:`capped collections <LINK TO MANUAL>`.

Use the ``TryNext()``method to create a tail-able cursor:

.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
   :language: go
   :dedent:
   :start-after: begin cursor try next
   :end-before: end cursor try next

For more information, see :manual:``Tailable Cursors <manual/core/tailable-cursors/>`` in the MongoDB manual.
```

Reply:
I don't think this is a capped collection. According to [our docs](https://docs.mongodb.com/manual/core/capped-collections/#tailable-cursor), you CAN use a tailable cursor with one, but not that it will by default. Additionally, [the spec](https://github.com/mongodb/specifications/blob/747b7480c47900827ff2b64f23c2a14636803d1a/source/crud/crud.rst#read) doesn't say anything about a capped collection, which makes me think it isn't or it's irrelevant to its operation.